### PR TITLE
Issue #1639 error:ordered comparison

### PR DESCRIPTION
**Pull request by damies13**
This is a fix for Issue #1639

### CODE REVIEW

**Comment by shun-iwasawa** [comment]
@damies13 Thank you for the fix - can you please squash the commits into one ?

**Comment by damies13** [comment]
> can you please squash the commits into one

@shun-iwasawa sure, can you teach me how to do that? (I am still learning how to use git)

**Comment by damies13** [comment]
I don't seem to have the necessary permissions to squash the commits, I tried to follow the github documentation on squashing commits.

From this page (https://help.github.com/articles/merging-a-pull-request/) it looks like you can do it when you merge.

**Comment by shun-iwasawa** [comment]
I previously wrote about git-based workflow to @gab3d [here](https://github.com/opentoonz/opentoonz/pull/1484#issuecomment-332717550).

To squash this PR, I think you can use `git rebase -i` command.
[Here](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) is a description for it.

So, in this PR how about doing like as follows:

1. Synchronize the local repository with the latest code from `upstream` 
    - Change the current branch to `master`
      `$ git checkout master`
    - Obtain the latest code
      `$ git pull upstream master`
    - Synchronize your `origin` with the local repository
      `$ git push origin master`

1. Rebase your commits to the current master
   - Move to the topic branch
     `$ git checkout Issue#1639-error-ordered-comparison`
   - Rebase
     `$ git rebase master`

1. Squash the commits
   - Check the commit log
     `git log --oneline`
      The history will be like as follows. In this case, there are 4 commits to be squashed.:
     ```
     6b78b9d your last commit
     db92104 commit3
     e4ae077 commit2
     44165a4 your first commit on the topic branch
     057de69 Allow Mesh Icon On Demand (#1604)                <------ current head of the main branch
     fc9220d Spanish UI - update to #1610 strings (#1633)
     ```
   - Start squashing your commits
     `$ git rebase -i 057de69` 
     Note that `057de69` should be replaced by the commit id of the current head of the main branch. 
     The following text will be shown in the editor:
     ```
     pick 44165a4 your first commit on the topic branch
     pick e4ae077 commit2
     pick db92104 commit3
     pick 6b78b9d your last commit
     ```

   - Replace `pick` by `s` for the commits to be squashed. Note that you should keep the first line unchanged.
     ```
     pick e4ae077 commit2              --> s e4ae077 commit2
     pick db92104 commit3             --> s db92104 commit3 
     pick 6b78b9d your last commit  --> s 6b78b9d your last commit 
     ```

   - Save and close the editor. Edit the commit message. Save again. 

1. Push your PR again
   `$ git push -f origin Issue#1639-error-ordered-comparison`
  using `-f` (force) option is needed because it will overwrite the history.

**Comment by shun-iwasawa** [comment]
@gab3d 
> From this page (https://help.github.com/articles/merging-a-pull-request/) it looks like you can do it when you merge.
Yes, I always use `Squash and merge` button. So you don't need to take care about the upstream history. Squashing your PR commits is just for make it easier to review.

**Comment by damies13** [comment]
@shun-iwasawa i'll give it a try tonight (i'm at work now)

File: toonz/sources/toonzqt/styleselection.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ bool pasteStylesDataWithoutUndo(TPalette *palette, TPaletteHandle *pltHandle,
       break;
     }
 
-    if (palette->getStylePage(styleId) < 0) {
+    if (palette->getStylePage(styleId) != NULL) {
```
Comment (shun-iwasawa) [inline toonz/sources/toonzqt/styleselection.cpp]: Should it be  `if (palette->getStylePage(styleId) == NULL) {` ?

I google-translated the Italian comment below.
If this condition is true, then the comment says; `styleId is not used: I use that. (cut / paste used to move styles)`
Whereas if this condition is false, the comment says; `styleId is already used. I have to take another one` .

**Comment by damies13** [comment]
@shun-iwasawa I tried following those steps for squashing but I think i made a big mess?

**Comment by shun-iwasawa** [comment]
I'll close this as this branch seems to fail squashing properly. Let's continue to #1663 .

**Pull request closed without merging**